common/djangoapps/student/forms.py

```diff
@@ -188,6 +188,19 @@ def __init__(
                                 "required": _("To enroll, you must follow the honor code.")
                             }
                         )
+                elif field_name == 'data_sharing_consent':
+                    if field_value == "required":
+                        self.fields[field_name] = TrueField(
+                            error_messages={
+                                "required": _(
+                                    "You must consent to data sharing to register."
```

Comment:
@srpearce, in this case, the message is seen when a direct call to the API is made, the `data_sharing_consent` field is required, and it's not present or set to False.

For the language that most users will see, here's a link to the relevant edx-enterprise code:

https://github.com/edx/edx-enterprise/blob/master/enterprise/templates/enterprise/grant_data_sharing_permissions.html